libs/cdk/utils/directives/readonly/fdk-readonly-provider.service.ts

```diff
@@ -32,17 +33,23 @@ export class FdkReadonlyProvider extends ReplaySubject<boolean> implements Reado
         @Optional() @SkipSelf() @Inject(FDK_READONLY_DIRECTIVE) private parentReadonly$: ReadonlyBehavior
     ) {
         super(1);
+
+        this._destroy$.onDestroy(() => {
+            this.complete();
+            this.readonlyObserver.unobserve(this.elementRef);
+        });
+
         combineLatest([this._readonlyChange$, this._viewModifiers$])
             .pipe(
                 tap(([isReadonly]) => this.setReadonlyState(isReadonly)),
-                takeUntil(this._destroy$)
+                takeUntilDestroyed(this._destroy$)
```

Comment:
is this._destroy$ needed here at all? It's injection context

Reply:
Since we already injected DestroyRef to use it's hook, why not pass it to the `takeUntilDestroyed`? It will save us time so that `takeUntilDestroyed` does not try to inject DestroyRef again.